### PR TITLE
Associate a file to a fileset

### LINKED ISSUE #369: Set type for generated file resource
When generated a resource for a deposited file, set `@type: http://sdr.sul.stanford.edu/models/sdr3-file.jsonld` in the metadata payload

### LINKED ISSUE #376: When uploading a file, associate it with a Fileset
See https://github.com/sul-dlss-labs/taco/pull/269/files#r179330969

We talked in standup 4/5/18 about using the `POST /resource/{fileset_id}/files` route rather than `/files` with a header to convey this information.

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #376 
Fixes #369

### CODE REVIEW

**Comment by jcoyne** [comment]
Yes, correct Aaron

**Comment by cmh2166** [comment]
@aaron-collier that is the docs related to administration, which we have.

File: validators/deposit_file.go [hunk 1]
```diff
@@ -12,12 +12,12 @@ type DepositFileValidator struct {
...
 // NewDepositFileValidator creates a new instance of DepositFileValidator
-func NewDepositFileValidator(repository db.Database) *DepositFileValidator {
+func NewDepositFileValidator(repository db.Database) ResourceValidator {
 	return &DepositFileValidator{repository: repository}
 }
 
-// ValidateResource validates that a headers are acceptable
-func (d *DepositFileValidator) ValidateResource(resource *multipart.FileHeader) error {
+// ValidateResource validates that the file resource is acceptable
+func (d *DepositFileValidator) ValidateResource(resource *datautils.Resource) *models.ErrorResponseErrors {
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply validators/deposit_file.go]: There is never a point at which a user submits JSON for a file deposit.  That JSON is generated by TACO.

File: handlers/deposit_file.go [hunk 2]
```diff
@@ -34,9 +39,12 @@ func (d *depositFileEntry) Handle(params operations.DepositFileParams, agent *au
 		return operations.NewDepositResourceUnauthorized()
 	}
 
-	validator := validators.NewDepositFileValidator(d.database)
-	if err := validator.ValidateResource(params.Upload.Header); err != nil {
-		return operations.NewDepositFileInternalServerError() // TODO: need a better error
+	upload := d.buildFile(params.Upload.Header, params.Upload.Data)
+	resource := d.buildPersistableResource(params.FilesetID, upload.Metadata)
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply handlers/deposit_file.go]: This is already handled by the `FileStructuralValidator`

File: examples/deposit_fileset.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  "label": "Schwend, Rudolf, Mayor of Zurich, ?-1391. Fileset 1.",
+  "access": {
+    "access": "world",
+    "download": "world"
+  },
+  "administrative": {
+    "created": "2012-10-28T04:13:43.639Z",
+    "deleted": false,
+    "isDescribedBy": "https://purl.stanford.edu/bq582kh2487.xml",
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply examples/deposit_fileset.json]: I've removed these fields so as not to cause undue confusion.

**Pull request merged**